app_dart/lib/src/request_handlers/github_webhook.dart

```diff
@@ -76,7 +63,7 @@ class GithubWebhook extends RequestHandler<Body> {
           final CheckRunEvent checkRunEvent = CheckRunEvent.fromJson(
             jsonDecode(stringRequest) as Map<String, dynamic>,
           );
-          await githubChecksService.handleCheckRun(checkRunEvent, luciBuildService);
+          await scheduler.processCheckRun(checkRunEvent);
```

Comment:
Thinking about moving forward to an eventual separation of the scheduler from the cocoon backend. Would it make sense to create a new webhook handler just for the scheduler?

Reply:
I think the proper fix is `github_webhook` shouldn't execute logic. The more logic we add to webhooks, the more likely a PR is to break all webhook functionality. We should be pushing the events to `TaskQueue` to be processed later (almost immediately, but we can retry if a failure occurs).

Reply:
The comment is more about separating logic in the webhook independently of how it is processed. Having logic in the webhook for label management, autosubmit bot, scheduling, etc makes it error prone causing someone adding a label processing logic to potentially break scheduling. Not a requirement for this PR but we should definitely start thinking about independent services for very focused functionality e.g a scheduler service just doing scheduling and nothing else.

Reply:
That makes sense, and I believe our concerns go hand in hand. With a microservice approach, we would still want some way to retry failed webhooks. The microservices would reduce the impact a bad change has to other parts of the system.